### PR TITLE
Fix Goroutine Leak

### PR DESCRIPTION
**Pull request by sfc-gh-jbahk**
### Description
Tests https://github.com/snowflakedb/gosnowflake/pull/650

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -186,6 +186,7 @@ func (sc *snowflakeConn) BeginTx(
 
 func (sc *snowflakeConn) cleanup() {
 	// must flush log buffer while the process is running.
+	sc.rest.Client.CloseIdleConnections()
```
Comment (sfc-gh-igarish) [inline connection.go:189]: Check for sc.rest != nil and sc.rest.Client != nil. This may be cause of NPE.

**Comment by sivchari** [comment]
I fixed test codes.

**Comment by sivchari** [comment]
@sfc-gh-jbahk ping

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

**Pull request merged**